app/src/main/java/edu/uc/hickmadc/customnotifications/ui/main/MainFragment.kt

```diff
@@ -32,6 +44,16 @@ class MainFragment : Fragment() {
         super.onActivityCreated(savedInstanceState)
         viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
 
+          val addButton = view?.findViewById<ImageButton>(R.id.btnAdd)
+                addButton?.setOnClickListener{
+                    var dialog= dialogfragment()
+
+                    dialog.show(requireActivity().supportFragmentManager!!, "customdialog")
+            }
+
+        val lView= view!!.findViewById<ListView>(R.id.listView)
+        val myListAdapter = MyListAdapter(activity!!,language,description,days, times)
+        lView.adapter = myListAdapter
```

Comment:
I also moved this code from the main activity class to the main fragment with the goal being increased modularization, and because of a few other changes I made. I moved the list view adapter code here because I also moved your data that you pass into the adapter here (see comment above) You can also see that I added the button click listener logic here, which was previously located in the main activity. Before your create new notification button was located on the main activity, which was causing it to be covered up by other ui components, and be located in places you probably didn't want it to be such as the settings tab. I moved your new notification button off of the main activity.xml file, and solely onto your main fragment. Now the button only appears on your main fragment/the homepage of your app.